<launch>
  <arg name="camera_name" default="camera" />
  <arg name="video_stream_provider" />
  <arg name="fps" default="30" />
  <arg name="frame_id" default="$(arg camera_name)" />
  <arg name="camera_info_url" default="" />
  <arg name="flip_horizontal" default="false" />
  <arg name="flip_vertical" default="false" />
  <arg name="width" />
  <arg name="height" />
  <arg name="frame_count" />
  <arg name="visualize" default="false"/>


  <group ns="$(arg camera_name)">
    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" />
    <node pkg="nodelet" type="nodelet" name="$(arg camera_name)_stream"
          args="load raw_video_stream/process_file camera_nodelet_manager">
      <remap from="camera" to="image_raw" />
      <param name="camera_name" type="string" value="$(arg camera_name)" />
      <param name="video_stream_provider" type="string" value="$(arg video_stream_provider)" />
      <param name="fps" type="int" value="$(arg fps)" />
      <param name="frame_id" type="string" value="$(arg frame_id)" />
      <param name="camera_info_url" type="string" value="$(arg camera_info_url)" />
      <param name="flip_horizontal" type="bool" value="$(arg flip_horizontal)" />
      <param name="flip_vertical" type="bool" value="$(arg flip_vertical)" />
      <param name="width" type="int" value="$(arg width)" />
      <param name="height" type="int" value="$(arg height)" />
      <param name="frame_count" type="int" value="$(arg frame_count)" />
    </node>

    <node if="$(arg visualize)" name="$(arg camera_name)_image_view" pkg="image_view" type="image_view">
      <remap from="image" to="image_raw" />
    </node>
  </group>

</launch>
